subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -2103,16 +2207,37 @@ static int handle_request(struct zoap_packet *request,
 		return -EINVAL;
 	}
 
+cleanup:
 	if (r == 0) {
```

Comment:
Since you removed the useless else {}, you could do:

if (r) {
   goto error;
}

and thus all the code block below  could be indented back. Nicer to read, less imbrication of if/else etc...